### PR TITLE
Menu

### PR DESCRIPTION
**Pull request by ScottyyyB**
## Description:
Implemented the Menu model and added the association to the Restaurant Model, also used jQuery for the smooth scrolling functionality. 

## What we learned:
* How to setup live-reload or as it's called 'guard-livereload' for rails
* How to setup jQuery for rails
* Smooth scrolling functionality using jQuery
* How to work with the seeds file add fake content, which is of course much better than having everyone create them in their own console.
* How to create model instances using FactoryBot when two models have an association
* How to properly set a supported version of Bootstrap

## Screenshot:

screen shot 2018-01-15 at 14 50 15

### CODE REVIEW

File: Guardfile [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+## Note: if you are using the `directories` clause above and you are not
+## watching the project directory ('.'), then you will want to move
+## the Guardfile to a watched dir and symlink it back, e.g.
+#
+#  $ mkdir config
+#  $ mv Guardfile config/
+#  $ ln -s config/Guardfile .
+#
+# and, you'll have to watch "config/Guardfile" instead of "Guardfile"
+
```
Comment (marizaswart) [inline Guardfile]: Remember to remove the comments

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+
+$(document).ready(function() {
+  var scrollLink = $('.scroll');
+
+  scrollLink.click(function(event) {
+    event.preventDefault();
+    $('body,html').animate({
+      scrollTop: $(this.hash).offset().top
+    }, 1500);
+  });
```
Comment (tochman) [inline app/assets/javascripts/application.js:14]: Is this the reason you are pulling in jQuery as a dependency?

File: features/step_definitions/basic_steps.rb [hunk 3]
```diff
@@ -22,8 +22,12 @@
   expect(current_path).to eq root_path
 end
 
+Then("show me the page") do
+  save_and_open_page
+end
+
```
Comment (marizaswart) [inline features/step_definitions/basic_steps.rb]: Remember to remove the show me the page as the last step :)

File: features/user_can_see_menu.feature [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  In order to help me decide what food to order
+  I would like to see Menus displayed
+
+  Background:
+    Given the following restaurants exists
+      | name            | description                                      |
+      | ThaiTanic       | Thailands finest food, watch out for the iceberg |
+      | Fu King Chinese | Best chinese food in town                        |
+    And the following menus exist for "ThaiTanic"
+      | name    |
```
Comment (SophieSjoberg) [inline features/user_can_see_menu.feature:12]: You should change name to menu_name just for clarification.

File: app/views/menus/_menu.html.erb [hunk 5]
```diff
@@ -0,0 +1,16 @@
+<div class='menu' id='menu-<%= menu.name %>'>
+  <%= link_to 'Back To Top', '#navigation-bar', class: 'scroll' %>
+  <h2 id="<%= menu.name %>"><%= menu.name %></h2>
```
Comment (tochman) [inline app/views/menus/_menu.html.erb]: You have 2 id’s set with menu name. Why? And could this be a problem? Also, why not use the menu.id so you’re 100% sure they are unique?

File: app/views/restaurants/show.html.erb [hunk 6]
```diff
@@ -1,9 +1,14 @@
-<h1><%= @restaurant.name %></h1>
+<h1 id='top'><%= @restaurant.name %></h1>
```
Comment (tochman) [inline app/views/restaurants/show.html.erb]: Can’t see this I’d being used. Am I just missing it?

**Review by SophieSjoberg — COMMENTED** [review]
LGTM except need to clarify name ; i e restaurant_name vs menu_name, otherwise it gets confusing.

**Review by marizaswart — COMMENTED** [review]
Great work, I made 2 minor remember comments :)

File: Guardfile (hunk 1 above)
Comment (NegativeKarma) [reply Guardfile]: Doing that

File: features/step_definitions/basic_steps.rb (hunk 3 above)
Comment (NegativeKarma) [reply features/step_definitions/basic_steps.rb]: Will do

File: features/user_can_see_menu.feature (hunk 4 above)
Comment (NegativeKarma) [reply features/user_can_see_menu.feature]: Ok, will do

File: app/assets/javascripts/application.js (hunk 2 above)
Comment (ScottyyyB) [reply app/assets/javascripts/application.js:14]: Yes it is

File: app/views/restaurants/show.html.erb (hunk 6 above)
Comment (ScottyyyB) [reply app/views/restaurants/show.html.erb]: Will look into this, think you're right forgot to remove this, was being used initially for the jQuery smooth scrolling, and I had this added for the 'Back To Top' links to scroll to the top of the page but I modified it to give the 'Back To Top' links the 'navigation-bar' id, so it actually scrolls to the top of the page. Will remove this 'top' id now.

File: app/views/menus/_menu.html.erb (hunk 5 above)
Comment (ScottyyyB) [reply app/views/menus/_menu.html.erb]: Forgot to remove the id on h2, done that now and changed the id on div to include menu id instead.

File: app/views/layouts/application.html.erb [hunk 7]
```diff
@@ -5,19 +5,21 @@
     <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
     <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.3/css/bootstrap.min.css" integrity="sha384-Zug+QiDoJOrZ5t4lssLdxGhVrurbmBWopoEl+M6BdEfwnCJZtKxi1KgxUyJq13dy" crossorigin="anonymous">
     <title>Food4U</title>
-    <%= csrf_meta_tags %>
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
```
Comment (diraulo) [inline app/views/layouts/application.html.erb:8]: Not a blocker but I'd prefer if this was loaded via the `bootstrap-sass` gem.

File: Guardfile [hunk 8]
```diff
@@ -0,0 +1,35 @@
+guard 'livereload' do
```
Comment (diraulo) [inline Guardfile:1]: Why are you setting up Guard as part of this PR? 🤔 

Not a blocker but please stay in scope when working on stories. I'd rather you create a separate chore to deal with this if you really needed it

File: features/user_can_see_menu.feature (hunk 4 above)
Comment (diraulo) [reply features/user_can_see_menu.feature:12]: Not necessary to change that to `menu_name` we're in the `Menu` domain so `name` is assumed to be the name of the menu

**Review by diraulo — COMMENTED** [review]
I have no objections re: Getting this merged. 

Just a few notes

Learn to **STAY IN SCOPE** when working on a feature. You've probably spent too much time setting up `guard-livereload` and Smooth scrolling functionality on this page.. Were those part of the requirements from your client/story? I highly doubt so. 

Doing things like this, even if it's beneficial for your personal learning, causes a burden to the overall's team progress on the project. You could've started on a different feature that actually adds real value to your client with the time spent on those 2 chores.

Please keep that in mind moving forward.

**Pull request merged**